### PR TITLE
gk-deploy: Deploy gluster-s3 container

### PR DESCRIPTION
**Pull request by jarrpa**
Depends on: #289, #305

---
This change is [Reviewable](https://reviewable.io/reviews/gluster/gluster-kubernetes/299)

### CODE REVIEW

**Comment by jarrpa** [comment]
Updated to remove non-s3 related fixes into #305.

**Comment by obnoxxx** [comment]
@jarrpa test fails

**Comment by jarrpa** [comment]
Error resolved, title updated.

**Comment by obnoxxx** [comment]
local complex test run passes for me if I include #323.
Planning to do full review tomorrow...

**Comment by humblec** [comment]
@obnoxxx did you get a chance to review this ?

**Comment by humblec** [comment]
@obnoxxx @jarrpa can we please get this reviewed/merged at the earliest?

**Comment by humblec** [comment]
@SaravanaStorageNetwork Can you please ack or nack this pr ?

File: deploy/gk-deploy [hunk 1]
```diff
@@ -110,6 +120,23 @@ Options:
...
+
+  --object-account ACCOUNT, --object-user USER, --object-password PASSWORD
+              Required credentials for deploying the gluster-s3 container. If
+              any of these are missing, object container deployment will be
+              skipped.
+
+  --object-sc STORAGE_CLASS
+              Specify an optional, pre-existing StorageClass for the object
+              container to use to create GlusterFS volumes to host the objects.
+              Default is to create a new StorageClass called '${OBJ_STORAGE_CLASS}'.
```
Comment (obnoxxx) [inline deploy/gk-deploy]: Could we make it a little more clear that the s3 container needs 2 volumes - one for the objects and one for the metadata? Like this it's a little confusing.

File: tests/complex/test-gk-deploy.sh [hunk 2]
```diff
@@ -9,4 +9,3 @@ copy_deploy
 run -r master -e "${TEST_DIR}/test-inside-gk-deploy.sh" "Test basic deployment"
 
 run -r master "${TEST_DIR}/test-inside-gk-deploy.sh" "Test basic deployment idempotence"
-
```
Comment (obnoxxx) [inline tests/complex/test-gk-deploy.sh]: This hunk does not seem to belong into this commit ... ;-)

**Comment by obnoxxx** [comment]
@jarrpa thanks for the update. the patches look good.
I want to run the tests locally once more and then merge.

**Comment by obnoxxx** [comment]
LGTM

**Pull request merged**